src/source/channel_router.rs

```diff
@@ -0,0 +1,306 @@
+// Channel router types and implementation.
+
+use crate::{Sample, Source};
+use std::{
+    cmp::min,
+    error::Error,
+    fmt,
+    sync::mpsc::{channel, Receiver, Sender},
+};
+
+/// A matrix to map inputs to outputs according to a gain
+///
+/// A two-dimensional matrix of `f32`s:
+/// - The first dimension is respective to the input channels
+/// - The second is respective to the output channels
+///
+/// Thus, if a value at `map[1,1]` is 0.2, this signifies that the signal on
```

Comment:
It is still confusing, I'd picked different numbers for input and output, or maybe just use names instead of constants (e.g. `map[in,out]`)